bitcoin/src/psbt/mod.rs

```diff
@@ -385,6 +397,79 @@ impl Psbt {
         Ok(used)
     }
 
+    /// Attempts to create all signatures required by this PSBT's `tap_key_origins` field, adding
+    /// them to `tap_key_sig` or `tap_script_sigs`.
+    ///
+    /// # Returns
+    ///
+    /// - Ok: A list of the public keys used in signing.
+    /// - Err: Error encountered trying to calculate the sighash AND we had the signing key.
+    fn bip32_sign_schnorr<C, K, T>(
+        &mut self,
+        k: &K,
+        input_index: usize,
+        cache: &mut SighashCache<T>,
+        secp: &Secp256k1<C>,
+    ) -> Result<Vec<PublicKey>, SignError>
+        where
+            C: Signing,
+            T: Borrow<Transaction>,
+            K: GetKey,
+    {
+
+        let mut input = self.inputs[input_index].clone();
+
+        let mut used = vec![]; // List of pubkeys used to sign the input.
+
+        for (xpk, (leaf_hashes, key_source)) in input.tap_key_origins.iter() {
+            let (sk, x_only_pubkey) = if let Ok(Some(sk)) = k.get_key(KeyRequest::Bip32(key_source.clone()), secp) {
+                (sk, *xpk)
+            } else {
+                continue;
+            };
+
+            //key path spend
+            if let Some(internal_key) = input.tap_internal_key {
+                if internal_key == x_only_pubkey && leaf_hashes.len() == 0 &&input.tap_key_sig.is_none() {
+                    let (msg, sighash_ty) = self.sighash_taproot(input_index, cache, None)?;
+                    let secp_for_tweak = &Secp256k1::<secp256k1::All>::gen_new();
```

Comment:
I don't think we want to do this. If we add `Verification` bound then we can use the function's `secp` argument.

Reply:
The reason why I did this before was because I didn’t want to modify the interface of `sign`, it would affect other function that call it.  Now I have modified it according to your suggestion and adjusted the scope of influence I found, I hope there will be no omissions.

Reply:
IIRC `Verification` will be always implied at some point but we need to support it in `secp256k1`.